src/web/page/templates.rs

```diff
@@ -1,45 +1,31 @@
 use crate::error::Result;
 use crate::web::rustdoc::RustdocPage;
-use crate::web::MetaData;
 use anyhow::Context;
 use rinja::Template;
-use std::{fmt, ops::Deref, sync::Arc};
+use std::{fmt, sync::Arc};
 use tracing::trace;
 
-macro_rules! rustdoc_page {
-    ($name:ident, $path:literal $(, $meta:ident)?) => {
-        #[derive(Template)]
-        #[template(path = $path)]
-        pub struct $name<'a> {
-            inner: &'a RustdocPage,
-        }
-
-        impl<'a> $name<'a> {
-            pub fn new(inner: &'a RustdocPage) -> Self {
-                Self { inner }
-            }
-
-            $(
-                pub(crate) fn $meta(&self) -> Option<&MetaData> {
-                    Some(&self.inner.metadata)
-                }
-            )?
-        }
-
-        impl<'a> Deref for $name<'a> {
-            type Target = RustdocPage;
+#[derive(Template)]
+#[template(path = "rustdoc/head.html")]
+pub struct Head<'a> {
+    rustdoc_css_file: &'a Option<String>,
```

Comment:
I would rather use `Option<&'a str>`, to omit the second indirection. (https://rust-lang.github.io/rust-clippy/stable/index.html#/ref_option_ref)